test/functional/api/search_api_test.rb

```diff
@@ -178,7 +178,7 @@ def app
   end
 
   test 'search Tag Nearby Nodes functionality with a valid query and specific period' do
-    get '/api/srch/taglocations?nwlat=171.0&selat=0.0&nwlng=0.0&selng=174.8&sort_by=recent&order_direction=ASC&to=2018-08-16'
+    get "/api/srch/taglocations?nwlat=171.0&selat=0.0&nwlng=0.0&selng=174.8&sort_by=recent&order_direction=ASC&to=2018-08-16" # Very weird Test failing here
```

Comment:
Yeah, this is failing, and I'm not very sure why. 
This is the error message:

```
ERROR SearchApiTest#test_search_Tag_Nearby_Nodes_functionality_with_a_valid_query_and_specific_period (131.05s)
Minitest::UnexpectedError:         ArgumentError: wrong number of arguments (given 2, expected 1)
            test/functional/api/search_api_test.rb:181:in `block in <class:SearchApiTest>'

```
No clue why this is happening.

Reply:
Okay, I'm seeing this error in multiple test suites now, and I decided to actually go to the pages that these endpoints are hitting. The Passenger logs reveals the error: 
```
Completed 500 Internal Server Error in 44ms (ActiveRecord: 12.7ms)
  
ActiveRecord::ConfigurationError (Can't join 'Comment' to association named 'answer'; perhaps you misspelled it?):
  
app/controllers/home_controller.rb:154:in `activity'

```

I believe fixing that function will help in fixing all the corresponding errors.

Reply:
Great, good solving work!